modules/common/vault-secrets-operator-setup/operator-setup.tf

```diff
@@ -41,26 +42,62 @@ resource "kubernetes_secret_v1" "vault-auth-default" {
   type = "kubernetes.io/service-account-token"
 }
 
+resource "null_resource" "download_vault" {
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+  #download vault binary
+  provisioner "local-exec" {
+    command = "curl -L -o vault.zip https://releases.hashicorp.com/vault/1.19.5/vault_1.19.5_linux_amd64.zip"
+  }
+  #unzip vault binary
+  provisioner "local-exec" {
+    command = "unzip vault.zip"
+  }
 
-resource "vault_auth_backend" "kubernetes" {
-  type = "kubernetes"
-  path = local.account_region_name
+  provisioner "local-exec" {
+    command = "chmod +x vault"
+  }
 }
 
-# vault read auth/indico-dev-us-east-2-dop-999/config
-resource "vault_kubernetes_auth_backend_config" "vault-auth" {
```

Comment:
Can't we just use the vault provider for this module? Define the provider only within this module. If the problem is that private connection clients can't access our vault, removing the provider from this module and adding manual commands wouldn't change anything. It may not work that way, but I believe that when you don't enable this vso module the provider inside the module isn't used either. I may have misunderstood the context of the task too

Reply:
No, we cannot. Terraform complains if you try to enable a provider within a submodule. That was a test I performed (commit history show it)

Reply:
That's sad, thanks for explaining that